ladybug/color.py

```diff
@@ -268,186 +322,192 @@ def therm(cls):
-            self._colors = Colorset()[1]
+            self._colors = Colorset.original()
         else:
-            assert hasattr(cols, "__iter__"), "Colors should be an iterable type"
+            assert isinstance(cols, Iterable) \
+                and not isinstance(cols, (str, dict, bytes, bytearray)), \
+                'Colors should be a list or tuple. Got {}'.format(type(cols))
             try:
-                cols = [col if isinstance(col, Color) else Color(
-                    col.R, col.G, col.B) for col in cols]
-            except Exception:
+                cols = tuple(col if isinstance(col, Color) else Color(
+                    col.R, col.G, col.B) for col in cols)
+            except AttributeError:
                 try:
-                    cols = [Color(col.Red, col.Green, col.Blue) for col in cols]
-                except Exception:
-                    raise ValueError("%s is not a vlid color" % str(cols))
+                    cols = tuple(Color(col.Red, col.Green, col.Blue) for col in cols)
+                except AttributeError:
+                    raise ValueError("{} is not a valid list of colors".format(cols))
+            if self._is_domain_set:
+                self.domain = self.domain  # re-check the domain against new colors
             self._colors = cols
 
     @property
-    def ctype(self):
-        """Chart type."""
-        return self._ctype
-
-    @ctype.setter
-    def ctype(self, t):
-        assert 0 <= int(t) <= 2, "Chart Type should be between 0-2\n" + \
-            "0: continuous, 1: segmented, 2: ordinal"
-        self._ctype = int(t)
+    def domain(self):
+        """Get or set the domain defining the color range."""
+        return self._domain
 
-    def color(self, value):
-        """Return color for an input value."""
-        assert self._is_domain_set, \
-            "Domain is not set. Use self.domain to set the domain."
+    @domain.setter
+    def domain(self, dom):
+        # check and prepare domain
+        if not dom:
+            dom = (0, 1)
+        else:
+            assert isinstance(dom, Iterable) \
+                and not isinstance(dom, (str, dict, bytes, bytearray)), \
+                'Domain should be a list or tuple. Got {}'.format(type(dom))
+            for val in dom:
+                assert isinstance(val, (float, int)), 'Values of a domain must be ' \
+                    'numbers. Got {}.'.format(type(val))
+            dom = sorted(map(float, dom))
+
+        if self._continuous_colors:  # continuous
+            # if type is continuous domain can only be 2 values
+            # or at least 1 value less than number of colors
+            if len(dom) == 2:
+                # remap domain based on colors
+                _step = float(dom[1] - dom[0]) / (len(self._colors) - 1)
+                _n = dom[0]
+                dom = tuple(_n + c * _step for c in range(len(self._colors)))
+
+            assert len(self._colors) >= len(dom), \
```

Comment:
You can see [here](https://github.com/ladybug-tools/ladybug/pull/141/files/dfce249ef88d2f4104e55a53ecc32884edd437fa#diff-604a419b183a1c82099ebc5480cb0241L390) that this whole section was from your original code and so that is why it's here now.  I agree that it doesn't make much sense and an `else` statement would be better.  I'll add it now.